src/lib/jobs/check_refunds.rs

```diff
@@ -63,49 +65,87 @@ pub async fn fetch_and_process_refunds(bq: BQClient, db_pool: &Pool<Postgres>) {
             continue;
         }
         // Do we already have it in the refunds table
-        let in_db = refunds.fetch_one_by_refund_id(&r.refund_id).await.is_ok();
-        if in_db {
-            match refunds.update_refund(&mut r).await {
-                Ok(_) => {
-                    println!("Refund {} updated. Continuing...", r.refund_id);
-                }
-                Err(e) => {
-                    // TODO - LOGGING
+        match refunds.fetch_one_by_refund_id(&r.refund_id).await {
+            Ok(mut refund) => {
+                // Only update if data is different
+                if refund.subscription_id == r.subscription_id
```

Comment:
This would mean creating a new struct, putting the existing data into the new struct, and then doing an equality comparison. If we were doing this comparison more than once I would say it's worth it. But I'm not sure it is. Although it would make this code more readable.